File: app/src/main/java/com/hoversoftsoln/esta_fort/core/repos/UserRepository.java
package com.hoversoftsoln.esta_fort.core.repos;

import com.hoversoftsoln.esta_fort.core.mvp.BasePresenter;
import com.hoversoftsoln.esta_fort.core.mvp.BaseView;

interface UserContracts {
    void getMyAccount();
}

public class UserRepository extends BasePresenter<UserRepository.View> implements UserContracts {

    public UserRepository() {
    }

    @Override
    public void getMyAccount() {

    }

    public interface View extends BaseView {

    }
}

File: app/src/main/java/com/hoversoftsoln/esta_fort/core/repos/UserRepository.kt
package com.hoversoftsoln.esta_fort.core.repos

import com.hoversoftsoln.esta_fort.core.data.User
import com.hoversoftsoln.esta_fort.core.mvp.BasePresenter
import com.hoversoftsoln.esta_fort.core.mvp.BaseView

internal interface UserContracts {
    fun getMyAccount()
}

class UserRepository : BasePresenter<UserRepository.View>(), UserContracts {

    override fun getMyAccount() {

    }

    interface View : BaseView
}
